volatility/framework/symbols/windows/extensions/network.py

```diff
@@ -0,0 +1,297 @@
+inaddr6_any = inet_ntop(socket.AF_INET6, [0] * 16)
+
+# copied to here as symbol space does not appear to support custom Enums
+TCP_STATE_ENUM = {
+    0: 'CLOSED',
+    1: 'LISTENING',
+    2: 'SYN_SENT',
+    3: 'SYN_RCVD',
+    4: 'ESTABLISHED',
+    5: 'FIN_WAIT1',
+    6: 'FIN_WAIT2',
+    7: 'CLOSE_WAIT',
+    8: 'CLOSING',
+    9: 'LAST_ACK',
+    12: 'TIME_WAIT',
+    13: 'DELETE_TCB'
+}
+
+class _TCP_LISTENER(objects.StructType):
+    """Class for objects found in TcpL pools.
+
+    This class serves as a base class for all pooled network objects.
+
+    It exposes some functions which return sanity-checked members. Substructures referred to by a
+    pointer may appear valid at first glance but will throw an InvalidAddressException on access.
+
+    This is not a problem when objects are validated via their `is_valid()` method, but when
+    scanning for semi-corrupted data this check will not be performed.
+
+    Be mindful that most of those methods return `None` when they would access invalid data.
+    If you want to process the raw data access the attributes directly, e.g.
+    via `network_object.InetAF` instead of `network_object.get_address_family()`.
+
+    """
+
+    def __init__(self, context: interfaces.context.ContextInterface, type_name: str,
+                 object_info: interfaces.objects.ObjectInformation, size: int,
+                 members: Dict[str, Tuple[int, interfaces.objects.Template]]) -> None:
+
+        super().__init__(context = context,
+                         type_name = type_name,
+                         object_info = object_info,
+                         size = size,
+                         members = members)
+
+    def get_address_family(self):
+        try:
+            return self.InetAF.dereference().AddressFamily
+
+        except exceptions.InvalidAddressException:
+            return None
+
+    def get_owner(self):
+        try:
+            return self.member('Owner').dereference()
+
+        except exceptions.InvalidAddressException:
+            return None
+
+    def get_owner_pid(self):
+        try:
+            if self.get_owner().is_valid():
+                return self.get_owner().UniqueProcessId
+            else:
+                return None
+
+        except exceptions.InvalidAddressException:
+            return None
+
+    def get_owner_procname(self):
+        try:
+            if self.get_owner().is_valid():
+                return self.get_owner().ImageFileName.cast(
+                        "string", max_length = self.get_owner().ImageFileName.vol.count, errors = "replace")
+            else:
+                return None
+
+        except exceptions.InvalidAddressException:
+            return None
+
+    def get_create_time(self):
+        dt_obj = conversion.wintime_to_datetime(self.CreateTime.QuadPart)
+
+        if isinstance(dt_obj, interfaces.renderers.BaseAbsentValue):
+            return dt_obj
+
+        # return None if the timestamp seems invalid
```

Comment:
Hmmmm, probably fine, since futuristic connections shouldn't exist, but limits like this make me nervous.

Reply:
Me too, but when testing some (invalid) objects returned stuff like the year 1600 and I don't want data like that to be recognized as valid.

Reply:
Yeah, that's fair enough.  Shouldn't really be possible to have tcp connections before 1950, and hopefully there'll be a newer version of volatility that people use in 150 years, so happy to mark this as resolved.  5:)  I'd also prefer this were pulled out clearly as a class variable (like `CORRUPT_DEFAULT`) though, just so it's very clear this happens and can be changed/fixed easily if it becomes necessary.  Mebbe I'm being too cautious.  Happy for you to resolve this however you'd like.  5:)

Reply:
Putting this into a class variable seems like a good idea... on the off chance that someone has set their system to 1650 (for whatver reason) this would make it easier for the investigator to change that value. But I don't deem it important enough to put it into a flag.

Reply:
Yep, I'd whole heartedly agree.  No need to expose it to the user, but good to pull it out and make it more visible for anyone that goes looking...

Reply:
updated. please check and resolve if ok :)